xarray/core/combine.py

```diff
@@ -703,7 +700,7 @@ def combine_by_coords(
 
     >>> x1
     <xarray.Dataset>
-    Dimensions:        (x: 3, y: 2)
+    Dimensions:        (y: 2, x: 3)
```

Comment:
should we worry about breaking downstream doctests?

Reply:
Yes, a bit, but on balance I think it's worth having the ordering in the repr be consistent with the ordering in the object...

Reply:
Do you expect many projects testing the repr?

Reply:
No, fairly rare I imagine